src/ExcelDataReader.DataSet/ExcelDataReaderExtensions.cs

```diff
@@ -107,7 +110,17 @@ private static DataTable AsDataTable(IExcelDataReader self, ExcelDataTableConfig
 
                         // if a column already exists with the name append _i to the duplicates
                         var columnName = GetUniqueColumnName(result, name);
-                        var column = new DataColumn(columnName, typeof(object)) { Caption = name };
+
+                        DataColumn column;
+                        if (configuration.UseColumnDataType)
+                        {
+                            column = new DataColumn(columnName, typeof(object)) { Caption = name };
+                        }
+                        else
+                        {
+                            column = new DataColumn(columnName, System.Type.GetType("System.String"));
```

Comment:
And what about { Caption = name }?

Reply:
Interesting. What do you propose the caption would be used for in the DataSet?

Reply:
I'm not sure, so the Caption isn't obliged? When You're initializing by passing typeof() it is used, but in GetType() not. So shouldn't we stick to one scenario?

Reply:
Ah, yes. I see. Looks like the prior code was _always_ using `Caption = name`  and that I was thinking if the `UseColumnDataType` was false, then the type would just be strings with no caption. I'm not sure why I might have left out the caption part. Perhaps without types, just keep the data clean?  (clearly I should have left a comment there)

Do you know how the caption value is manifested in Excel and/or the XML generated? Perhaps as a tooltip / hover text? I'm not familiar with that.

I don't think it is a required field (or if it is, it has a default) - as I have this code working daily in production.

I agree it should be consistent. Should I modify the PR? I won't have time to test this in the near future.

Reply:
It should have been consistent if we were to merge this, but given the discussion in #446 I think rather to close instead.